sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3494,6 +3494,46 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       Seq(Row(Map[Int, Int]()), Row(Map(1 -> 2))))
   }
 
+  test("SPARK-31334: Don't ResolveReference/ResolveMissingReference when " +
+    "Filter condition with aggregate expression") {
+    Seq(
+      (1, 3),
+      (2, 3),
+      (3, 6),
+      (4, 7),
+      (5, 9),
+      (6, 9)
+    ).toDF("a", "b").createOrReplaceTempView("testData1")
+
+    checkAnswer(sql(
```

Comment:
does this test fail before your patch?

Reply:
> does this test fail before your patch?

No, it's won't failed, here is for contrast.

Reply:
So this test is not qualified to reproduce the bug?

Reply:
> So this test is not qualified to reproduce the bug?

The first SQL is used for comparison, and the second can reproduce bugs.
If don't need, we can just delete first one.

Reply:
let's delete

Reply:
> let's delete

Done